extensions/harvest/src/listTimeEntries.tsx

```diff
@@ -0,0 +1,186 @@
+            actions={
+              <ActionPanel>
+                <ActionPanelSection title={`${entry.project.name} | ${entry.client.name}`}>
+                  <ToggleTimerAction onComplete={init} entry={entry} />
+                  {/* Disabling Edit Action for now so we can ship something a useable extension faster */}
+                  {/* <EditEntryAction onSave={init} entry={entry} /> */}
+                  <DeleteEntryAction onComplete={init} entry={entry} />
+                </ActionPanelSection>
+                <ActionPanelSection title="Harvest">
+                  <NewEntryAction onSave={init} />
+                  {harvestInstallPath && <OpenHarvestAppAction app={harvestInstallPath} />}
+                </ActionPanelSection>
+              </ActionPanel>
+            }
+          />
+        );
+      })}
+    </List>
+  );
+}
+
+function NewEntryAction({
+  onSave = async () => {
+    return;
+  },
+}: {
+  onSave: () => Promise<void>;
+}) {
+  return (
+    <PushAction
+      target={<New onSave={onSave} />}
+      title="New Time Entry"
+      shortcut={{ key: "n", modifiers: ["cmd"] }}
+      icon={Icon.Plus}
+    />
+  );
+}
+
+function EditEntryAction({
+  onSave = async () => {
+    return;
+  },
+  entry,
+}: {
+  onSave: () => Promise<void>;
+  entry: HarvestTimeEntry;
+}) {
+  return (
+    <PushAction
+      target={<New onSave={onSave} entry={entry} />}
+      title="Edit Time Entry"
+      shortcut={{ key: "e", modifiers: ["cmd"] }}
+      icon={Icon.Pencil}
+    />
+  );
+}
+
+function DeleteEntryAction({
+  onComplete = async () => {
+    return;
+  },
+  entry,
+}: {
+  onComplete: () => Promise<void>;
+  entry: HarvestTimeEntry;
+}) {
+  return (
+    <PushAction
+      target={<Delete onComplete={onComplete} entry={entry} />}
+      title="Delete Time Entry"
+      shortcut={{ key: "delete", modifiers: ["cmd"] }}
+      icon={Icon.Trash}
+    />
+  );
+}
+
+function ToggleTimerAction({ entry, onComplete }: { entry: HarvestTimeEntry; onComplete: () => Promise<void> }) {
+  return (
+    <ActionPanelItem
+      title={entry.is_running ? "Stop Timer" : "Start Timer"}
+      icon={Icon.Clock}
+      onAction={async () => {
+        const toast = new Toast({ style: ToastStyle.Animated, title: "Loading..." });
+        await toast.show();
+        try {
+          if (entry.is_running) {
+            await stopTimer(entry);
+          } else {
+            await restartTimer(entry);
+          }
+          await toast.hide();
+        } catch {
+          await showToast(ToastStyle.Failure, "Error", `Could not ${entry.is_running ? "stop" : "start"} your timer`);
+        }
+        await onComplete();
+      }}
+    />
+  );
+}
+
+function OpenHarvestAppAction({ app }: { app: Application }) {
+  return (
+    <ActionPanelItem
+      onAction={() => {
+        try {
+          execSync(`open ${app.path}`);
+        } catch {
+          showToast(ToastStyle.Failure, "Could not Open Harvest App");
```

Comment:
Here as a feature for the future would be nice to open web site if there is no app installed 😉

Reply:
Ah yes good idea!